### PR TITLE
Add startup logs

### LINKED ISSUE #4951: [V1] Investigate placeholder initialization errors
Sometimes initializing V1 placeholders fails with error below. If a customer function app lands on a placeholder that failed to initialize, function app stops working:

```
<Event><System><Provider Name="ASP.NET 4.0.30319.0"/><EventID>1309</EventID><Level>2</Level><Task>0</Task><Keywords>Keywords</Keywords><TimeCreated SystemTime="2019-08-04T01:00:38Z"/><EventRecordID>353494687</EventRecordID><Channel>Application</Channel><Computer>RD0003FF4C09D8</Computer><Security/></System><EventData><Data>3005</Data><Data>An unhandled exception has occurred.</Data><Data>8/4/2019 1:00:37 AM</Data><Data>8/4/2019 1:00:37 AM</Data><Data>56aa9386da834d818587e51780860163</Data><Data>2</Data><Data>1</Data><Data>0</Data><Data>/LM/W3SVC/1899347586/ROOT-12107-132093540360036456</Data><Data>Full</Data><Data>/</Data><Data>D:\Program Files (x86)\SiteExtensions\Functions\1.0.12582\</Data><Data>RD0003FF4C09D8</Data><Data/><Data>2620</Data><Data>w3wp.exe</Data><Data>IIS APPPOOL\mawsFnPlaceholder156_v1</Data><Data>HttpException</Data><Data>
   at System.Web.HttpApplicationFactory.EnsureAppStartCalledForIntegratedMode(HttpContext context, HttpApplication app)
   at System.Web.HttpApplication.RegisterEventSubscriptionsWithIIS(IntPtr appContext, HttpContext context, MethodInfo[] handlers)
   at System.Web.HttpApplication.InitSpecial(HttpApplicationState state, MethodInfo[] handlers, IntPtr appContext, HttpContext context)
   at System.Web.HttpApplicationFactory.GetSpecialApplicationInstance(IntPtr appContext, HttpContext context)
   at System.Web.Hosting.PipelineRuntime.InitializeApplication(IntPtr appContext)
   at Autofac.Core.Resolving.InstanceLookup.Activate(IEnumerable`1 parameters)
   at Autofac.Core.Resolving.InstanceLookup.Execute()
   at Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, IComponentRegistration registration, IEnumerable`1 parameters)
   at Autofac.Core.Resolving.ResolveOperation.Execute(IComponentRegistration registration, IEnumerable`1 parameters)
   at Autofac.Core.Lifetime.LifetimeScope.ResolveComponent(IComponentRegistration registration, IEnumerable`1 parameters)
   at Autofac.Core.Container.ResolveComponent(IComponentRegistration registration, IEnumerable`1 parameters)
   at Autofac.ResolutionExtensions.TryResolveService(IComponentContext context, Service service, IEnumerable`1 parameters, Object&amp; instance)
   at Autofac.ResolutionExtensions.ResolveOptionalService(IComponentContext context, Service service, IEnumerable`1 parameters)
   at Autofac.Integration.WebApi.AutofacWebApiDependencyScope.GetService(Type serviceType)
   at Autofac.Integration.WebApi.AutofacWebApiDependencyResolver.GetService(Type serviceType)
   at Microsoft.Azure.WebJobs.Script.WebHost.DependencyResolverExtensions.GetService[TService](IDependencyResolver resolver) in C:\projects\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\Extensions\DependencyResolverExtensions.cs:line 12
   at Microsoft.Azure.WebJobs.Script.WebHost.WebApiApplication.&lt;&gt;c__DisplayClass0_0.&lt;Application_Start&gt;b__0(HttpConfiguration c) in C:\projects\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\Global.asax.cs:line 34
   at System.Web.Http.GlobalConfiguration.Configure(Action`1 configurationCallback)
   at Microsoft.Azure.WebJobs.Script.WebHost.WebApiApplication.Application_Start() in C:\projects\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\Global.asax.cs:line 34
   at System.Net.NetworkInformation.SystemIPGlobalProperties.GetFixedInfo()
   at System.Net.NetworkInformation.SystemIPGlobalProperties.get_FixedInfo()
   at System.Net.NetworkInformation.SystemIPGlobalProperties.get_DomainName()
   at System.Net.CookieContainer..ctor()
   at System.Net.Http.HttpClientHandler..ctor()
   at Microsoft.Azure.WebJobs.Script.WebHost.WebScriptHostManager..ctor(ScriptHostConfiguration config, ISecretManagerFactory secretManagerFactory, IScriptEventManager eventManager, ScriptSettingsManager settingsManager, WebHostSettings webHostSettings, ILoggerFactory loggerFactory, IScriptHostFactory scriptHostFactory, ISecretsRepositoryFactory secretsRepositoryFactory, HostPerformanceManager hostPerformanceManager, Int32 hostTimeoutSeconds, Int32 hostPollingIntervalMilliseconds) in C:\projects\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\WebScriptHostManager.cs:line 48
   at Microsoft.Azure.WebJobs.Script.WebHost.WebHostResolver.EnsureInitialized(WebHostSettings settings) in C:\projects\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\App_Start\WebHostResolver.cs:line 142
   at Microsoft.Azure.WebJobs.Script.WebHost.WebHostResolver.GetActiveInstance[TInstance](WebHostSettings settings, TInstance&amp; activeInstance, TInstance&amp; standbyInstance) in C:\projects\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\App_Start\WebHostResolver.cs:line 294
   at Microsoft.Azure.WebJobs.Script.WebHost.AutofacBootstrap.&lt;&gt;c__DisplayClass0_0.&lt;Initialize&gt;b__4(IComponentContext ct) in C:\projects\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\App_Start\AutofacBootstrap.cs:line 33
   at Autofac.RegistrationExtensions.&lt;&gt;c__Di</Data><Data>http://sla-ws-func-prod-am2-119-orchtrigger-001.azurewebsites.net/admin/host/ping</Data><Data>/admin/host/ping</Data><Data>IPV4_ADDRESS_REDACTED</Data><Data/><Data>False</Data><Data/><Data>IIS APPPOOL\mawsFnPlaceholder156_v1</Data><Data>20</Data><Data>IIS APPPOOL\mawsFnPlaceholder156_v1</Data><Data>False</Data><Data>
   at System.Web.HttpApplication.RegisterEventSubscriptionsWithIIS(IntPtr appContext, HttpContext context, MethodInfo[] handlers)
   at System.Web.HttpApplication.InitSpecial(HttpApplicationState state, MethodInfo[] handlers, IntPtr appContext, HttpContext context)
   at System.Web.HttpApplicationFactory.GetSpecialApplicationInstance(IntPtr appContext, HttpContext context)
   at System.Web.Hosting.PipelineRuntime.InitializeApplication(IntPtr appContext)
</Data></EventData></Event>
```
- Need to improve error handling in placeholder warmup code path to give more details why the application failed to initialize
- Need to follow up on any platform improvements to avoid converting failed placeholder sites to customer apps
cc @fabiocav

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #4951

### CODE REVIEW

File: src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs [hunk 1]
```diff
@@ -123,74 +124,85 @@ public WebHookReceiverManager GetWebHookReceiverManager(WebHostSettings settings
         /// </summary>
         internal void EnsureInitialized(WebHostSettings settings)
         {
-            lock (_syncLock)
+            ILoggerFactory defaultLoggerFactory = GetDefaultLoggerFactory(settings);
+            _logger = defaultLoggerFactory.CreateLogger(ScriptConstants.LogCategoryHostStartup);
```
Comment (alrod) [inline src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs]: Create the logger in constructor?

File: src/WebJobs.Script.WebHost/WebScriptHostManager.cs [hunk 2]
```diff
@@ -453,6 +453,7 @@ private async void OnSyncTimerTick(object state)
 
         private void InitializeHttp()
         {
+            _logger.LogDebug("Initializing HttpRequestManager");
```
Comment (alrod) [inline src/WebJobs.Script.WebHost/WebScriptHostManager.cs]: Does this break some tests on relying on start up logs?

File: src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs (hunk 1 above)
Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs]: Creating GetDefaultLoggerFactory depends on 'WebHostSettings settings' which is not a DI dependency for `WebHostResolver`
@fabiocav - is it ok to update dependencies for `WebHostResolver` ?

File: src/WebJobs.Script.WebHost/WebScriptHostManager.cs (hunk 2 above)
Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/WebScriptHostManager.cs]: Assuming existing tests are ok as CI build is green.

**Pull request merged**